database/apikeys.go

```diff
@@ -150,3 +231,86 @@ func (db *DB) APIKeyList(ctx context.Context, user User) ([]APIKeyRecord, error)
 	}
 	return aks, nil
 }
+
+// APIKeyUpdate updates an existing API key. This works by replacing the
+// list of Skylinks within the API key record. Only valid for public API keys.
+func (db *DB) APIKeyUpdate(ctx context.Context, user User, akID primitive.ObjectID, skylinks []string) error {
+	if user.ID.IsZero() {
+		return errors.New("invalid user")
+	}
+	// Validate all given skylinks.
+	for _, s := range skylinks {
+		if !ValidSkylinkHash(s) {
+			return ErrInvalidSkylink
+		}
+	}
+	filter := bson.M{
+		"_id":     akID,
+		"public":  &True, // you can only update public API keys
```

Comment:
I never really understood the `&True` stuff, why don't you do `{$eq: true}`

Reply:
Believe me, I would love to. But the types used in MongoDB's go driver define the field as `*bool`, so I need to define a variable and give it a pointer to that. If you know of a nice way to avoid that, I would be happy to hear it.

Reply:
this is true but only in the case of option structs I think, in filters you can do `bson.M{"$eq": true}`

Reply:
Oh, you are right. Inertia, I guess. :)